### ISSUE TITLE
Ambiguous definitions: footnotes without spacing

### ISSUE THREAD

**Issue by yaycmyk** [wooorm/remark#106]
using `mdast.parse()`:

```js
mdast.parse("foo[^abc] bar\n\n[^abc]: Baz", {footnotes: true});
```

The "Baz" portion actually is created as a regular `definition` and the corresponding reference at `foo[^abc]` is a `linkReference`.

However, if I add a second word to the definition, that fixes it and `footerReference` & `footerDefinition` are created as expected. e.g.

```js
mdast.parse("foo[^abc] bar\n\n[^abc]: Baz baz", {footnotes: true});
```

So it seems like there might be a flaw in the parsing for footnotes that requires a sentence structure.

**Comment by wooorm** [comment]
Hey @yaycmyk. Thanks for raising the issues!

I’d like to take another look at the footnote grammar after version 3 (GH-96, GH-78). It will probably be removed from the core and into an external plug-in.

**Comment by wooorm** [comment]
This will be included in the upcoming release of mdast, 3.0.0, which I’ll expect to :ship: before the new year.

**Comment by yaycmyk** [comment]
:tada:
